study1/week3/reduce-array-size-to-the-half.java

```diff
@@ -0,0 +1,44 @@
+//https://leetcode.com/problems/reduce-array-size-to-the-half/description/
+
+class Solution {
+    public int minSetSize(int[] arr) {
+        //time complexity is O(n)
+        //space complexity is also O(n)
+        
+        int arrLength = arr.length;
```

Comment:
Rather making arrLength a variable, making halfArray would seem to make more sense as it's being used multiple times.
`int halfLength = arr.length / 2;`